src/test_utils/mod.rs

```diff
@@ -26,9 +26,36 @@ near_sdk_sim::lazy_static_include::lazy_static_include_bytes! {
 pub(crate) const SUBMIT: &str = "submit";
 
 pub(crate) mod erc20;
+pub(crate) mod self_destruct;
 pub(crate) mod solidity;
 pub(crate) mod standard_precompiles;
 
+pub(crate) struct Signer {
```

Comment:
Nice! I like this a lot. Might be worth refactoring other tests to use this in the future.